cbdr/cortex
# COR 689: Switch Asset File Naming Structure

**Purpose:** @justin3thompson  has requested for V2 Cortex, we switch from using the `ContentItem`'s guid to using a parameterized version of the asset's title. 


**JIRA:**
[COR-689](https://cb-content-enablement.atlassian.net/browse/COR-689)

**Changes:**
* Changes to setup
  * must have gemfile point cortex plugins to [#refactor/COR-689/Switch-Asset-File-Naming-Structure](https://github.com/cortex-cms/cortex-plugins-core/tree/refactor/COR-689/Switch-Asset-File-Naming-Structure)

* Architectural changes
  * `ContentItemService` now has a class_attribute `:form_fields` this is so the FieldType subclasses can access their fellow field's data since ContentItem and other FieldItems have yet to be created. 
 * Media `ContentType` has a new metadata field `:naming_data`

* Migrations
  * must reseed: `bundle exec rake cortex:core:db:reseed`
  
* Library changes
  * a new version of `cortex-plugins-core` will need to be created after this

* Side effects
  * I need to think about this one

**Screenshots**
* Before
 n/a

* After
n/a

**QA Links:**
n/a

**How to Verify These Changes**
* Specific pages to visit
  * n/a

* Steps to take
  * create a media content type
  * click edit to see the URL it should have a snake case version of the asset name. 
  * edit the title to make sure the asset url slug does not change

* Responsive considerations
  * n/a


**Relevant PRs/Dependencies:**
[https://github.com/cortex-cms/cortex-plugins-core/pull/49](https://github.com/cortex-cms/cortex-plugins-core/pull/49)

**Additional Information**
n/a
